src/common/commonutils/UserUtils.c

```diff
@@ -819,49 +816,37 @@ int CheckNoDuplicateUidsExist(char** reason, OsConfigLogHandle log)
     return status;
 }
 
-int RemoveUser(SimplifiedUser* user, bool removeHome, OsConfigLogHandle log)
+static int RemoveUser(const char* username, const uid_t uid, const gid_t gid, const char* home, OsConfigLogHandle log)
 {
-    const char* commandTemplate = "userdel %s %s";
-    char* command = NULL;
-    int status = 0, _status = 0;
+    const char* commandTemplate = "userdel -f %s";
+    char command[64] = {0};
+    int status = 0;
+    int commandLen = 0;
 
-    if (NULL == user)
+    if (NULL == username)
     {
         OsConfigLogError(log, "RemoveUser: invalid argument");
         return EINVAL;
     }
-    else if (0 == user->userId)
+
+    commandLen = snprintf(command, sizeof(command), commandTemplate, username);
+    if(((int)sizeof(command) <= commandLen) || (0 > commandLen))
     {
-        OsConfigLogInfo(log, "RemoveUser: cannot remove user with uid 0 ('%s' %u, %u)", user->username, user->userId, user->groupId);
-        return EPERM;
+        OsConfigLogError(log, "RemoveUser: command buffer too small");
```

Comment:
>command buffer too small"); [](http://example.com/codeflow?start=43&length=27)

Add how many bytes the buffer is, and how many bytes are necessary. So we will understand the bug from the trace. E.g.:

OsConfigLogError(log, "RemoveUser: command buffer length of %lu bytes is under required length of %ly bytes");